src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpResponseMessageHelper.cs

```
@@ -0,0 +1,367 @@
+            bool hasContent = await ValidateContentTypeAsync();
+            Message message = null;
+
+            if (!hasContent)
+            {
+                if (_encoder.MessageVersion == MessageVersion.None)
+                {
+                    message = new NullMessage();
+                }
+                else
+                {
+                    return null;
+                }
+            }
+            else
+            {
+                message = await ReadStreamAsMessageAsync();
+            }
+
+            var exception = ProcessHttpAddressing(message);
+            Contract.Assert(exception == null, "ProcessHttpAddressing should not set an exception after parsing a response message.");
+
+            return message;
+        }
+
+        private Exception ProcessHttpAddressing(Message message)
+        {
+            Exception result = null;
+            AddProperties(message);
+
+            // check if user is receiving WS-1 messages
+            if (message.Version.Addressing == AddressingVersion.None)
+            {
+                bool actionAbsent = false;
+                try
+                {
+                    actionAbsent = (message.Headers.Action == null);
+                }
+                catch (XmlException)
+                {
+                }
+                catch (CommunicationException)
+                {
+                }
+                if (!actionAbsent)
+                {
+                    result = new ProtocolException(SR.Format(SR.HttpAddressingNoneHeaderOnWire,
+                        XD.AddressingDictionary.Action.Value));
+                }
+
+                bool toAbsent = false;
+                try
+                {
+                    toAbsent = (message.Headers.To == null);
+                }
+                catch (XmlException)
+                {
+                }
+                catch (CommunicationException)
+                {
+                }
```

There are several catch/ignores here around seemingly innocuous property sets.  Could you add comments what's going on and why these exceptions can be ignored?